cluster/aws/templates/format-disks.sh

```diff
@@ -183,7 +193,9 @@ if [[ -n "${move_docker}" ]]; then
     mv /var/lib/docker ${move_docker}/
   fi
   mkdir -p ${move_docker}/docker
-  ln -s ${move_docker}/docker /var/lib/docker
+  if [[ ! -e /var/lib/docker ]]; then
```

Comment:
Yeah, messy... I'll add a comment to explain.

We want `/var/lib/docker` to be a symlink to `/mnt/ephemeral/docker`.  Line 193 only fires if it is a directory (`-d`).  If you stop and start your instance, that corrupts /mnt/ephemeral (ephemeral disks) but the root volume is usually persisted (if on EBS).

So the problem we're solving here is that /var/lib/docker may already be a symlink to /mnt/ephemeral/docker, but /mnt/ephemeral may have been wiped.